integrations/api_keys_test.go

```diff
@@ -37,3 +39,54 @@ func TestDeleteDeployKeyNoLogin(t *testing.T) {
 	req := NewRequest(t, "DELETE", "/api/v1/repos/user2/repo1/keys/1")
 	MakeRequest(t, req, http.StatusUnauthorized)
 }
+
+func TestCreateReadOnlyDeployKey(t *testing.T) {
```

Comment:
Thank you very much for adding tests!

However, our suite of integration tests is becoming quite unwieldy. Would you consider instead writing equivalent unit tests for the `DeployKeysPost`handler? See [routers/repo/issue_label_test.go](https://github.com/go-gitea/gitea/blob/master/routers/repo/issue_label_test.go) for some examples.

Reply:
I wrote the unit tests, but I don't know if it's ok to dump the integration tests.
Those are testing the API and the unit tests the interface.

Reply:
Okay, let's leave the integration tests. Eventually, it might make sense to convert our API integration tests to unit tests, but that's for another day. Thanks for adding unit tests!